modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageExpirationWorker.java

```diff
@@ -62,8 +81,72 @@ public void run() {
         while (true) {
             if (working) {
                 try {
+
+                    //delete the accumulated messages in the list that are filtered out from the message flusher expiration rule
+                    //should be run in all the nodes
+                    if (!expiredMessageSet.isEmpty()) {
+                        MessagingEngine.getInstance().deleteMessages(new ArrayList<AndesMessageMetadata>(expiredMessageSet));
+                    }
+
+                    boolean isClusterEnabled = AndesContext.getInstance().isClusteringEnabled();
+
+                    if ( isClusterEnabled && AndesContext.getInstance().getClusterAgent().isCoordinator()) {
```

Comment:
This whole section needs to be removed, right? It's the same code we have in 'DBBasedExpiryMessageDeletionTask' isn't it?